app/DashBoard/Presenters/HomePagePresenter.php

```diff
@@ -18,26 +18,92 @@ class HomePagePresenter extends BasePresenter
 	private $projectsRepository;
 
 
+	/**
+	 * @var \Pd\Monitoring\UsersFavoriteProjects\UsersFavoriteProjectsRepository
+	 */
+	private $usersFavoriteProjectsRepository;
+
+    /**
+     * @var \Pd\Monitoring\User\User
+     */
+    private $userRepository;
+
+    private $favoriteArray = [];
```

Comment:
V maximální možné míře prosím typuj všechno, co jen jde. Vždy je to alespoň pole. Potom `array|User[]`. IDE si s tím poradí